src/api/encoding.cc

```diff
@@ -133,20 +135,30 @@ enum encoding ParseEncoding(Isolate* isolate,
   return ParseEncoding(*encoding, default_encoding);
 }
 
+MaybeLocal<Value> TryEncode(Isolate* isolate,
+                            const char* buf,
+                            size_t len,
+                            enum encoding encoding) {
+  CHECK_NE(encoding, UCS2);
```

Comment:
```suggestion
  DCHECK_NE(encoding, UCS2);
```

Reply:
This part was taken from the existiing impl and I'd prefer to keep it the same. We can change this separately if appropriate.